src/rules/requireJsdocRule.ts

```diff
@@ -0,0 +1,135 @@
+                    type: "boolean",
+                },
+                methodDefinition: {
+                    type: "boolean",
+                },
+            },
+            additionalProperties: false,
+        },
+        optionExamples: ['[true, {"exportedOnly": false, "classDeclaration": false}]',
+                         '[true, {"exportedOnly": false, "classDeclaration": false, functionDeclaration: false, methodDefinition: false}]'],
+
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING = "Missing JSDoc comment.";
+
+    public applyWithProgram(sourceFile: ts.SourceFile, langSvc: ts.LanguageService): Lint.RuleFailure[] {
+        const noForInArrayWalker = new RequireJSDocWalker(sourceFile, this.getOptions(), langSvc.getProgram());
+        return this.applyWithWalker(noForInArrayWalker);
+    }
+}
+
+class RequireJSDocWalker extends Lint.ProgramAwareRuleWalker {
+    public visitClassDeclaration(node: ts.ClassDeclaration) {
+        const settings = this.getRequireJSDocOptions();
+        if (settings.classDeclaration) {
+            this.checkJSDoc(node);
+        }
+        super.visitClassDeclaration(node);
+    }
+
+    public visitFunctionExpression(node: ts.FunctionExpression) {
+        const settings = this.getRequireJSDocOptions();
+        if (settings.functionDeclaration) {
+            this.checkJSDoc(node);
+        }
+        super.visitFunctionExpression(node);
+    }
+
+    public visitFunctionDeclaration(node: ts.FunctionDeclaration) {
+        const settings = this.getRequireJSDocOptions();
+        if (settings.functionDeclaration) {
+            this.checkJSDoc(node);
+        }
+        super.visitFunctionDeclaration(node);
+    }
+
+    public visitMethodDeclaration(node: ts.MethodDeclaration) {
+        const settings = this.getRequireJSDocOptions();
+        if (settings.methodDefinition) {
+            this.checkJSDoc(node);
+        }
+        super.visitMethodDeclaration(node);
+    }
+
+    private getRequireJSDocOptions() {
+        const DEFAULT_OPTIONS = {
+            classDeclaration: true,
+            exportedOnly: true,
+            functionDeclaration: true,
+            methodDefinition: true,
+        };
+
+        const ruleOptions = this.getOptions();
+        return Object.assign(DEFAULT_OPTIONS, ruleOptions && ruleOptions[0]);
+    }
+
+    private checkJSDoc(node: ts.FunctionDeclaration | ts.FunctionExpression | ts.ClassDeclaration | ts.MethodDeclaration) {
+        if (node.name) {
+            const checker = this.getTypeChecker();
+            const symbol = checker.getSymbolAtLocation(node.name);
+            const hasDocs = symbol.getDocumentationComment().length > 0;
+
+            if (!hasDocs ) {
+                const settings = this.getRequireJSDocOptions();
+                // console.log(settings);
+                if (settings.exportedOnly && !this.isExportedNode(node)) {
```

Comment:
you should check if you want to enforce JSDoc on this node *before* you use the typechecker